packages/@aws-cdk/custom-resource-handlers/lib/aws-stepfunctions-tasks/cross-region-aws-sdk-handler/index.ts

```diff
@@ -0,0 +1,79 @@
+/* eslint-disable @typescript-eslint/no-require-imports */
+/* eslint-disable no-console */
+
+type Event = {
+  /**
+   * AWS Region name to call the service
+   */
+  region: string;
+
+  /**
+   * AWS service name (must be in AWS SDK for JS v3 style)
+   */
+  service: string;
+
+  /**
+   * AWS API endpoint
+   */
+  endpoint: string;
+
+  /**
+   * API Action name (must be in camelCase or PascalCase)
+   */
+  action: string;
+
+  /**
+   * request parameters to call the API (must be in AWS SDK for JS v3 style)
+   */
+  parameters?: any;
+};
+
+// reference: code from CDK AwsCustomResource
+function findV3ClientConstructor(pkg: object) {
+  const [_clientName, ServiceClient] = Object.entries(pkg).find(([name]) => {
+    // Services expose a base __Client class that we don't want ever
+    return name.endsWith('Client') && name !== '__Client';
+  }) as [
+    string,
+    {
+      new (config: any): {
+        send: (command: any) => Promise<any>;
+        config: any;
+      };
+    },
+  ];
+  return ServiceClient;
+}
+
+function findCommandClass(pkg: object, action: string) {
+  const camelToPascal = (str: string) => str.charAt(0).toUpperCase() + str.slice(1);
+  const commandName = `${camelToPascal(action)}Command`;
+  const Command = Object.entries(pkg).find(([name]) => name.toLowerCase() === commandName.toLowerCase())?.[1] as {
+    new (input: any): any;
+  };
+  if (!Command) {
+    throw new Error(`Unable to find command named: ${commandName} for action: ${action} in service package ${pkg}`);
+  }
+  return Command;
+}
+
+export const handler = async (event: Event) => {
+  // Log the event to understand the incoming request structure
+  console.log('Event: ', event);
+
+  try {
+    // esbuild-disable unsupported-require-call -- not esbuildable but that's fine
+    const pkg = require(`@aws-sdk/client-${event.service}`);
```

Comment:
Correct me if I'm wrong, but `normalizeServiceName` is not being ran for your handler? It would allow us to keep the same pattern as other `service`s props, see [AwsSdkCall](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.custom_resources.AwsSdkCall.html#service):

https://github.com/aws/aws-cdk/blob/fff9cf694b14811682c8671a1e55afa53151df8b/packages/@aws-cdk/aws-custom-resource-sdk-adapter/lib/api-call.ts#L65

https://github.com/aws/aws-cdk/blob/fff9cf694b14811682c8671a1e55afa53151df8b/packages/@aws-cdk/aws-custom-resource-sdk-adapter/lib/sdk-info.ts#L10-L15

Reply:
I'd like to keep this feature and API as simple as possible to lower the maintenance cost. `AwsSdkCall` code has become such complicated for backward compatibility reason. Since this feature is new, we don't have to introduce the complexity at all.

Also the API is almost compatible with the existing `CallAwsService` construct, which also does not support normalizing service names.

Reply:
I'll let the maintainer weigh in, I think it's a nice piece of QoL, but I understand your maintainability concern

Reply:
To help me understand, can you elaborate on your concern? What maintenance cost are you concerned with using AwsSdkCall code.

Reply:
AwsSdkCall's code is mainly to solve the v2-v3 compatibility problem. Since this construct does not require such compatibility, I think there is little advantage to using the existing code. After all, the new Lambda is only code under 100 lines. Also, if we want to reuse the existing code, we need to refactor it first (for example, currently it's only for CFn custom resource handler, but this one is for invocation from SFn). Given the ROI, I'd rather keep them independent.